analyzers/src/SonarAnalyzer.VisualBasic/Rules/ParametersCorrectOrder.cs

```diff
@@ -88,11 +88,26 @@ protected override void Initialize(SonarAnalysisContext context)
         protected override Location GetMethodDeclarationIdentifierLocation(SyntaxNode syntaxNode) =>
             (syntaxNode as MethodBlockBaseSyntax)?.FindIdentifierLocation();
 
-        protected override SyntaxToken? GetArgumentIdentifier(ArgumentSyntax argument) =>
-            (argument.GetExpression() as IdentifierNameSyntax)?.Identifier;
+        protected override SyntaxToken? GetArgumentIdentifier(ArgumentSyntax argument) => GetExpressionSyntaxIdentifier(argument?.GetExpression());
```

Comment:
The original line was `(argument.GetExpression() as IdentifierNameSyntax)?.Identifier;` so the `?` shouldn't be needed. Or am I missing something and you have a failing test case for this? (I didn't run the UTs yet)

```suggestion
        protected override SyntaxToken? GetArgumentIdentifier(ArgumentSyntax argument) => GetExpressionSyntaxIdentifier(argument.GetExpression());
```